### DISCUSSION TITLE
Autofix Formatting analyzer

### DISCUSSION THREAD

**Discussion by larocca1** [JosefPihrt/Roslynator#851] [Q&A]
Hello. I have installed the Formatting Analyzer nuget package and configure a .editorconfig file with warning for some formatting analysers. Is there a way in visual studio to automatically fix all the formatting warnings for a document?
Regards

**Comment by josefpihrt** [comment]
I'm not aware of any built-in functionality.

Nevertheless, Roslynator CLI is designed exactly for this purpose.

Please follow the [instructions](https://github.com/JosefPihrt/Roslynator/tree/master/docs/cli) how to use it.

If you have .NET Core projects it should be sufficient to install the tool:
```
dotnet tool install roslynator.dotnet.cli -g
```

and then the run the tool:
```
roslynator fix <PATH_TO_SOLUTION_OR_PROJECT>
```

Also you can set working directory to the directory where the sln/csproj is located and then run just:
```
roslynator fix
```
